opacus/privacy_engine.py

```diff
@@ -68,18 +68,12 @@ def forbid_accumulation_hook(
         return
 
     for _, p in trainable_parameters(module):
-        if p.grad_sample is not None:
-            if isinstance(p.grad_sample, torch.Tensor):
-                accumulated_iterations = 1
-            elif isinstance(p.grad_sample, list):
-                accumulated_iterations = len(p.grad_sample)
-
-            if accumulated_iterations > 1:
```

Comment:
Is this change a consequence of moving to `register_backward_hook ` ?

Reply:
Indeed. 

`register_backward_hook` hooks are called after submodule-level hooks, while `register_full_backward_hook` is called before we compute per sample gradients for the current iteration

Reply:
That said, I'm looking at the failing GAN example and I'm not so sure about this anymore